unit_test/test_arguments.py

```diff
@@ -0,0 +1,188 @@
+import unittest
+import nttt
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+class TestArguments(unittest.TestCase):
+
+    def test_get_final_step(self):
+        ''' Test case for the get_final_step function:
+        - If there are no step files, the function should return 0.
+        - If there is one step file, called step_1.md, the function should return 1.
+        - If there are two step files, called step_1.md and step_2.md, the function should return 2.
+        - If there are three step files, called step_1.md, step__2.md and step_4.md, the function should return 2.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            result = nttt.arguments.get_final_step(temp_folder)
+            self.assertEqual(result, 0)
+
+            step_file = nttt.arguments.get_step_file(temp_folder, 1)
```

Comment:
this test is a bit hard to read because it's very imperative. Several methods from the class under the tests are used which may lead to false-positives.

Consider creating the step files explicitly in the simplest possible way (like `Path(dir, "step_1.md").touch()`). This may improve readability.

Reply:
Yes, now using the Path class to create step files directly.